gui/widgets/label.py

```diff
@@ -39,7 +44,17 @@ def show(self):
         super().show()  # Draw or erase border
         wri = self.writer
         dev = self.device
-        Writer.set_textpos(dev, self.row, self.col)
+        if self.align == ALIGN_LEFT:
+          Writer.set_textpos(dev, self.row, self.col)
+        else:
+          txt_width = wri.stringlen(txt)
+          if self.width <= txt_width:
```

Comment:
I think the logic here is wrong. If `self.width <= txt_width` it should behave as per the released code, i.e. left align. You are centre aligning.

Reply:
I think it is correct. After right indenting it looks
```
        if self.align == ALIGN_LEFT:
            Writer.set_textpos(dev, self.row, self.col)
        else:
            txt_width = wri.stringlen(txt)
            if self.width <= txt_width:
                Writer.set_textpos(dev, self.row, self.col)
            else:
                if self.align == ALIGN_RIGHT:
                    Writer.set_textpos(dev, self.row, self.col + self.width - txt_width)
                else:
                    Writer.set_textpos(dev, self.row, self.col + self.width // 2 - txt_width // 2)

```
Line 6 of the fragment (width of string is greater or equal then the width of the label) is identical to the line 2, which is the original one.